Compilation instructions: CBMC must be built using Java 8

The instructions in `COMPILING.md` say to install Debian package `openjdk-8-jdk` (or `openjdk8` on FreeBSD), but that is not sufficient.  It must also be the default Java version.  If JDK 8 is installed, but the default is JDK 25, then under Ubuntu the build fails with:

```
[ERROR] COMPILATION ERROR : 
[ERROR] cbmc/jbmc/lib/java-models-library/src/main/java/java/lang/Object.java:[40,7] flexible constructors is not supported in -source 8
  (use -source 25 or higher to enable flexible constructors)
[ERROR] cbmc/jbmc/lib/java-models-library/src/main/java/java/lang/Class.java:[199,24] cannot find symbol
  symbol:   method checkClassLoaderPermission(java.lang.ClassLoader,<nulltype>)
  location: class java.lang.ClassLoader
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.6.1:compile (default-compile) on project models: Compilation failure: Compilation failure: 
[ERROR] cbmc/jbmc/lib/java-models-library/src/main/java/java/lang/Object.java:[40,7] flexible constructors is not supported in -source 8
[ERROR]   (use -source 25 or higher to enable flexible constructors)
[ERROR] cbmc/jbmc/lib/java-models-library/src/main/java/java/lang/Class.java:[199,24] cannot find symbol
[ERROR]   symbol:   method checkClassLoaderPermission(java.lang.ClassLoader,<nulltype>)
[ERROR]   location: class java.lang.ClassLoader
[ERROR] -> [Help 1]

--- a/COMPILING.md
+++ b/COMPILING.md
@@ -167,18 +167,15 @@ We assume that you have a Debian/Ubuntu or Red Hat-like distribution.
    ```
    dnf install java-1.8.0-openjdk-devel maven jq
    ```
-   JBMC requires Java 8 to be the active default JDK; installing it alongside
-   a newer JDK is not sufficient. Verify both the runtime and compiler before
-   building:
+   JBMC requires a Java 8 toolchain in the build shell, but Java 8 does not
+   need to be the system-wide default JDK. If another JDK is selected by
+   default, set `JAVA_HOME` to the Java 8 installation and prepend its `bin`
+   directory to `PATH` for the build. Verify both the runtime and compiler:
    ```
    java -version
    javac -version
    ```
-   Both commands should report version 1.8. If another JDK is selected, use
-   `update-alternatives --config java` and
-   `update-alternatives --config javac` on Debian-like systems, or
-   `alternatives --config java` and `alternatives --config javac` on
-   Red Hat/Fedora systems, and select Java 8.
+   Both commands should report version 1.8.
 
 2. As a user, get the CBMC source via
    ```
@@ -262,10 +259,10 @@ Maven 3 manually.
    ```
    pkg install openjdk8 wget maven
    ```
-   Make sure the Java 8 installation is active before building. Set
-   `JAVA_HOME` to the OpenJDK 8 directory and prepend its `bin` directory to
-   `PATH` if another JDK is selected, then verify with `java -version` and
-   `javac -version`.
+   Java 8 needs only be selected for the build shell, not as the system-wide
+   default. Set `JAVA_HOME` to the OpenJDK 8 directory and prepend its `bin`
+   directory to `PATH` if another JDK is selected, then verify with
+   `java -version` and `javac -version`.
 2. As a user, get the CBMC source via
    ```
    git clone https://github.com/diffblue/cbmc cbmc-git

--- a/COMPILING.md
+++ b/COMPILING.md
@@ -167,6 +167,18 @@ We assume that you have a Debian/Ubuntu or Red Hat-like distribution.
    ```
    dnf install java-1.8.0-openjdk-devel maven jq
    ```
+   JBMC requires Java 8 to be the active default JDK; installing it alongside
+   a newer JDK is not sufficient. Verify both the runtime and compiler before
+   building:
+   ```
+   java -version
+   javac -version
+   ```
+   Both commands should report version 1.8. If another JDK is selected, use
+   `update-alternatives --config java` and
+   `update-alternatives --config javac` on Debian-like systems, or
+   `alternatives --config java` and `alternatives --config javac` on
+   Red Hat/Fedora systems, and select Java 8.
 
 2. As a user, get the CBMC source via
    ```
@@ -250,6 +262,10 @@ Maven 3 manually.
    ```
    pkg install openjdk8 wget maven
    ```
+   Make sure the Java 8 installation is active before building. Set
+   `JAVA_HOME` to the OpenJDK 8 directory and prepend its `bin` directory to
+   `PATH` if another JDK is selected, then verify with `java -version` and
+   `javac -version`.
 2. As a user, get the CBMC source via
    ```
    git clone https://github.com/diffblue/cbmc cbmc-git

